scripts/ndjson_to_csv/ndjson_to_csv.py

```diff
@@ -0,0 +1,122 @@
+# /usr/bin/python3
+
+import csv, json, sys, tempfile
+
+from functools import reduce
+from pathlib import Path
+
+import pandas as pd
+from tqdm import tqdm
+
+
+col_headers = set()
+help_message = 'Convert FHIR data stored in an NDJSON file to a CSV format.\n\
+\n\
+Usage:\n\
+\n\
+    ndjson_to_csv.py <inputfile> <output_filename>\n\
+\n\
+\n\
+<inputfile>        Source file to be converted (.json, .ndjson, .txt)\n\
+<output_filename>  Optional desired output filename\n\
+\n\
+-h, --help         view help message\n'
+
+def flatten(data, parent_key='', sep='.'):
+    items = []
+    for k, v in data.items():
+        new_key = parent_key + sep + k if parent_key else k
+        if isinstance(v, dict):
+            items.extend(flatten(v, new_key, sep).items())
+        elif isinstance(v, list):
+            if not isinstance(v[0], (str, int)):
+                if len(v) > 1:
+                    for index, item in zip(range(len(v)), v):
+                        if index > 0:
+                            items.extend(flatten(
+                                item,
+                                '{}{}'.format(new_key, index+1),
+                                sep
+                            ).items())
+                        else:
+                            items.extend(flatten(item, new_key, sep).items())
+                else:
+                    for item in v:
+                        items.extend(flatten(item, new_key, sep).items())
+            else:
+                list_strings = [str(item) for item in v]
+                col_headers.add(new_key)
+                items.append((new_key, ' '.join(list_strings)))
+        else:
+            col_headers.add(new_key)
```

Comment:
I think the program might read better if you initialize `col_headers` in the main, then after `flatten` is invoked just add the headers.
```
data = flatten(json.loads(line))
col_headers.update(data.keys())
tf.write(f'{json.dumps(data)}\n')
```

Reply:
Good thought

Reply:
included.